src/formatting/closures.rs

```diff
@@ -103,6 +103,18 @@ pub(crate) fn rewrite_closure(
     }
 }
 
+// Check if closure block wrapping may not be preserved (#4394)
+fn can_try_rewrite_without_block(body: &ast::Expr, context: &RewriteContext<'_>) -> bool {
+    if !context.inside_macro()
+        && (!context.config.preserve_closure_block_wrapping()
+            || !context.snippet(body.span).trim().starts_with('{'))
+    {
+        true
+    } else {
+        false
+    }
+}
```

Comment:
I don't know that this necessarily needs to be a module-level function; an internal function or closure within `rewrite_closure` would probably have been better, though not necessarily a blocker. 

Remember that the primary purpose of this is to improve readability and maintainability, so there's no requirement to do this check within a single control flow expression.:

I think this could be expressed in a more readable way such as something like:
```rust
if context.inside_macro() {
    return false;
}
if context.config.preserve_closure_block_wrapping() && context.snippet(body.span).trim_start().starts_with('{') {
  return false;
}
true
```

Also, there's really no need to do anything with the end of the snippet, so prefer `trim_start` vs. `trim`. Another option (without utilizing a separate function like this), would be to create a local a variable immediately before the `match fn_decl.output` match expression and use that as the arm guard since the check is used as a guard on the very first arm anyway.

I really just want to try to avoid having a single guard be so complicated relative to the entire match expression itself

Reply:
I made the change using local variable for the condition.  I hope it is good enough now.

Reply:
> (I see that several of the auto-tests failed, but as far as I understand this is because of test environment issues and not problem in the submitted code, so I am not re-submitting the changes.)

Yup, no worries. We have a broken toolstate issue at the moment that will prevent compilation with recent nightlies. Once I get that sorted will come back to this to re-start the CI checks